### ISSUE TITLE
BUG: nullable schema

### ISSUE THREAD

**Issue by ssssota** [Himenon/openapi-typescript-code-generator#91]
Hi! I think this project is useful and so interesting.

I have a question. Why do generator use interface for schema definition?

I think type alias is more useful.

e.g.

https://swagger.io/specification/#:~:text=A%20true%20value%20adds%20%22null%22%20to%20the%20allowed%20type%20specified%20by%20the%20type%20keyword%2C%20only%20if%20type%20is%20explicitly%20defined%20within%20the%20same%20Schema%20Object.

```yaml
schema:
  type: object
  nullable: true
  properties:
    foo:
      type: integer
```

```ts
type schema = { foo: number } | null;
```

**Comment by Himenon** [comment]
It does not matter whether it is a Type Alias or an interface. I believe the essence of the issue is whether the problem is solved or not.

By the way, do you mean that there is a problem with specifying `nullable` in the case that @ssssota  wrote about? Could you please provide some more details?

**Comment by Himenon** [comment]
And this is not what we talk about in the topic of Type Alias or Interface. What you really want to do with the issue is to put it in the title. Otherwise, I will close it. 

@ssssota

**Comment by ssssota** [comment]
> I believe the essence of the issue is whether the problem is solved or not.

I think so too.

---

I'm using [OpenAPITools/openapi-generator](https://github.com/OpenAPITools/openapi-generator) to generate types and client. And I am considering where to migrate to.

I have schemas like follows:

```yaml
components:
  schemas:
    member:
      type: object
      additionalProperties: false
      required:
        - child
      properties:
        child:
          $ref: "#/components/schemas/child"
    child:
      type: object
      nullable: true
      additionalProperties: false
      required:
        - name
      properties:
        name:
          type: string
```

_https://github.com/ssssota/openapi-typescript-code-generator/blob/e1b63b408604def8686187698316eead3f8f5d1b/test/nullable.schema.domain/index.yml#L18-L36_

This generator generates like follows:

```ts
    export interface member {
        child: Schemas.child;
    }
    export interface child {
        name?: string;
    }
```

_https://github.com/ssssota/openapi-typescript-code-generator/blob/6b3e5f573102dfcd9f0ffed4582f56df7719756f/test/__tests__/__snapshots__/nullable-schema-domain-test.ts.snap#L13-L18_

I think it is not correct.
So, I patched with type-alias and generate like follows:

```ts
    export type member = {
        child: Schemas.child;
    };
    export type child = {
        name: string;
    } | null;
```

_https://github.com/ssssota/openapi-typescript-code-generator/blob/e1b63b408604def8686187698316eead3f8f5d1b/test/__tests__/__snapshots__/nullable-schema-domain-test.ts.snap#L14-L19_

This is why I asked the previous question.

> And this is not what we talk about in the topic of Type Alias or Interface. What you really want to do with the issue is to put it in the title. Otherwise, I will close this issue.

OK. I'll change title.

**Comment by ssssota** [comment]
> is it a situation where the schema cannot be changed?

Maybe no.
However, I don't want to take that workaround.

**Comment by Himenon** [comment]
I published [v0.19.2](https://github.com/Himenon/openapi-typescript-code-generator/releases/tag/%40himenon%2Fopenapi-typescript-code-generator%400.19.2). Thanks your bug report !

And, I could align everything to type alias, but that would be a disruptive change to existing users and I need to provide options to minimize that. However, I do not have the time to provide that option, so I will only release a patch to allow nullable.
